### PR TITLE
DSD-91: Image component lazy loading

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-91](https://jira.nypl.org/browse/DSD-91)

## This PR does the following:

- Adds lazy loading for images in the `Image` component with the help of the `react-intersection-observer` and `@charlietango/use-native-lazy-loading` npm packages.
- First, if the browser supports the native `loading="true"` `img` attribute, then use it. Otherwise, the IntersectionObserver API is used.

## How has this been tested?

- In Storybook through the "Network" tab in Chrome's console window for the `Card`, `SimpleGrid`, `Hero`, `Image`, `StructuredContent`, and `Table` components.
- Note that tests that use the `Image` component internally need the `IntersectionObserver` API to be mocked with the `mockAllIsIntersecting(true);` call.

## Accessibility concerns or updates

- N/A.

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

File: src/components/Card/Card.test.tsx [hunk 1]
```diff
@@ -345,8 +359,10 @@ describe("Card", () => {
     expect(screen.getByText(/Read Online/i)).toBeInTheDocument();
   });
 
-  it("Generates a card without an image block if no image is provided", () => {
+  it("Generates a card with the full-click functionality", () => {
```
Comment (EdwinGuzman) [inline src/components/Card/Card.test.tsx:362]: Updated this title because it was a copy/paste.

File: src/components/Hero/Hero.test.tsx [hunk 2]
```diff
@@ -23,68 +24,73 @@ const imageProps = {
...
-        <Hero
-          heroType="primary"
-          heading={<Heading level="one" id="a11y-hero" text="Hero Primary" />}
-          subHeaderText="Example Subtitle"
-          backgroundImageSrc="https://placeimg.com/1600/800/arch"
-        />
-      );
-      expect(await axe(container)).toHaveNoViolations();
-    });
+describe("Hero accessbility tests", () => {
```
Comment (EdwinGuzman) [inline src/components/Hero/Hero.test.tsx:27]: Moved accessibility tests to a separate "describe".

File: src/components/Image/Image.stories.mdx [hunk 3]
```diff
@@ -330,3 +331,42 @@ in an element with a fixed width value.
...
+
+The `Image` component handles lazy loading images through two npm packages:
+[`react-intersection-observer`](https://github.com/thebuilder/react-intersection-observer)
+and [`@charlietango/use-native-lazy-loading`](https://www.npmjs.com/package/@charlietango/use-native-lazy-loading).
+The first package handles using the `IntersectionObserver` API in React and the
+second pacakge checks if the `img`'s native `loading="lazy` attribute is
+supported by the browser.
+
+The `Image`'s implementation of lazy loading is based on the recipe provided by
+them. Once the image's wrapper is in view, we load the image by rendering it
```
Comment (bigfishdesign13) [inline src/components/Image/Image.stories.mdx]: Provided by who?  What does `them` refer to?  The two packages?

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
This is working well, but I would like to make one additional change.  Add a background color to the `img` element to act as a placeholder as image files load.

```
backgroundColor: "ui.gray.x-light-cool”,
```

**Comment by EdwinGuzman** [comment]
Updated. It's hard to see the background gray color placeholder but I slowed down the network and took some screenshots:
Screen Shot 2022-06-14 at 2 23 20 PM
Screen Shot 2022-06-14 at 2 23 33 PM

So, unless if you do this or have slow internet connection, it'll be very subtle.

**Review by bigfishdesign13 — APPROVED** [review]
Very nice.

**Pull request merged**